modules/3d/perf/perf_rendering.cpp

```diff
@@ -0,0 +1,267 @@
+                        0, 1.0,   0, -position[1],
+                        0,   0, 1.0, -position[2],
+                        0,   0,   0,          1.0);
+    res = res * translate;
+
+    return res;
+}
+
+template<typename T>
+std::string printEnum(T v)
+{
+    std::string s;
+    std::stringstream ss;
+    v.PrintTo(&ss);
+    ss >> s;
+    return s;
+}
+
+// resolution, shading type, outputs needed
+typedef perf::TestBaseWithParam<std::tuple<std::tuple<int, int>, ShadingTypeEnum, OutputsEnum, GlCompatibleModeEnum>> RenderingTest;
+
+PERF_TEST_P(RenderingTest, rasterizeTriangles, ::testing::Combine(
+    ::testing::Values(std::make_tuple(1920, 1080), std::make_tuple(1024, 768), std::make_tuple(640, 480)),
+    ShadingTypeEnum::all(),
+    OutputsEnum::all(),
+    GlCompatibleModeEnum::all()
+    ))
+{
+    auto t = GetParam();
+    auto wh = std::get<0>(t);
+    int width = std::get<0>(wh);
+    int height = std::get<1>(wh);
+    auto shadingType = std::get<1>(t);
+    auto outputs = std::get<2>(t);
+    auto glCompatibleMode = std::get<3>(t);
+
+    string objectPath = findDataFile("rendering/spot.obj");
+
+    Vec3d position = Vec3d( 2.4, 0.7, 1.2);
+    Vec3d lookat   = Vec3d( 0.0, 0.0, 0.3);
+    Vec3d upVector = Vec3d( 0.0, 1.0, 0.0);
+
+    double fovy = 45.0;
+
+    std::vector<Vec3f> vertices;
+    std::vector<Vec3i> indices;
+    std::vector<Vec3f> colors;
+    std::vector<vector<int>> indvec;
+    loadMesh(objectPath, vertices, colors, indvec);
+    for (const auto &vec : indvec)
+    {
+        indices.push_back({vec[0], vec[1], vec[2]});
+    }
+
+    if (shadingType == TriangleShadingType::White)
+    {
+        colors.clear();
+    }
+    else
+    {
+        for (auto &color : colors)
+        {
+            color = Vec3f(abs(color[0]), abs(color[1]), abs(color[2]));
+        }
+    }
+
+    double zNear = 0.1, zFar = 50.0;
+
+    Matx44d cameraPose = lookAtMatrixCal(position, lookat, upVector);
+    double fovYradians = fovy * (CV_PI / 180.0);
+    TriangleRasterizeSettings settings;
+    settings.setCullingMode(TriangleCullingMode::CW)
+            .setShadingType(shadingType)
+            .setGlCompatibleMode(glCompatibleMode);
+
+    Mat depth_buf, color_buf;
+    while (next())
+    {
+        // Prefilled to measure pure rendering time w/o allocation and clear
+        float zMax = (glCompatibleMode == TriangleGlCompatibleMode::InvertedDepth) ? 1.f : (float)zFar;
+        depth_buf = Mat(height, width, CV_32F, zMax);
+        color_buf = Mat(height, width, CV_32FC3, Scalar::all(0));
+
+        startTimer();
+        cv::triangleRasterize(vertices, indices, colors, cameraPose, fovYradians, zNear, zFar, settings,
+                              (outputs != Outputs::ColorOnly) ? depth_buf : noArray(),
+                              (outputs != Outputs::DepthOnly) ? color_buf : noArray());
+        stopTimer();
+    }
+
+    if (debugLevel > 0)
+    {
+        cvtColor(color_buf, color_buf, cv::COLOR_RGB2BGR);
```

Comment:
Is it  BGR intentionally?

Reply:
See the comment [above](https://github.com/opencv/opencv/pull/24459#discussion_r1473650323)